generate_template.sh

```diff
@@ -6,67 +6,80 @@ set -u
 
 module list
 
-ENVIRONMENT_PARAMETERS=parameters_gattaca.csv
-RAWDATANAME=${1}
-WORKDIR=$2
-GROUP=$3
-WORKFLOW=${EBROOTNGS_DEMULTIPLEXING}/workflow.csv
+ENVIRONMENT_PARAMETERS="parameters_gattaca.csv"
+RAWDATANAME="${1}"
+WORKDIR="${2}"
+GROUP="${3}"
+WORKFLOW="${EBROOTNGS_DEMULTIPLEXING}/workflow.csv"
 echo "$WORKDIR AND $RAWDATANAME"
 echo "GROUPIE: $GROUP"
 
-if [ -f .compute.properties ];
+if [ -f ".compute.properties" ];
 then
-     rm .compute.properties
+     rm ".compute.properties"
 fi
 
-mkdir -p ${WORKDIR}/generatedscripts/${RAWDATANAME}/
-mkdir -p ${WORKDIR}/rawdata/ngs/${RAWDATANAME}/
-if [ -f ${WORKDIR}/generatedscripts/${RAWDATANAME}/out.csv  ];
+mkdir -p "${WORKDIR}/generatedscripts/${RAWDATANAME}/"
+mkdir -p "${WORKDIR}/rawdata/ngs/${RAWDATANAME}/"
+
+if [ -f "${WORKDIR}/generatedscripts/${RAWDATANAME}/out.csv"  ];
 then
-    	rm -rf ${WORKDIR}/generatedscripts/${RAWDATANAME}/out.csv
+	rm -rf "${WORKDIR}/generatedscripts/${RAWDATANAME}/out.csv"
 fi
 
 #
 ###### Dual barcode checker
 #
-sampsheet=${WORKDIR}/generatedscripts/${RAWDATANAME}/${RAWDATANAME}.csv
+sampsheet="${WORKDIR}/generatedscripts/${RAWDATANAME}/${RAWDATANAME}.csv"
+
+mac2unix "${sampsheet}"
 
-mac2unix ${sampsheet}
+declare -a _sampleSheetColumnNames=()
+declare -A _sampleSheetColumnOffsets=()
 
-rm -f barcode2.isthere
+IFS="," read -r -a _sampleSheetColumnNames <<< "$(head -1 "${sampsheet}")"
+	for (( _offset = 0 ; _offset < ${#_sampleSheetColumnNames[@]:-0} ; _offset++ ))
+	do
+		_sampleSheetColumnOffsets["${_sampleSheetColumnNames[${_offset}]}"]="${_offset}"
+	done
+dualBarcode="unknown"
+if [[ -n "${_sampleSheetColumnOffsets["barcode"]+isset}" ]]; then
+	barcodeFieldIndex=$((${_sampleSheetColumnOffsets["barcode"]} + 1))
+	readarray -t barcodes< <(tail -n +2 "${sampsheet}" | cut -d "," -f "${barcodeFieldIndex}" | sort | uniq )
+	for barcode in "${barcodes[@]}"
+	do
+		if [[ "${barcode}" == *"-"* ]]
+		then
+			echo "TRUE"
+			dualBarcode="TRUE"
+		else
+			echo "false"
```

Comment:
Capital letters or not? The echo says: TRUE & false. Please use: TRUE & FALSE or true & false